client/landing/gutenboarding/stores/domain-suggestions/selectors.ts

```diff
@@ -1,15 +1,63 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
+import { STORE_KEY } from '.';
 import { DomainSuggestionQuery } from './types';
 import { State } from './reducer';
-import { getSerializedDomainsSuggestionsQuery } from 'state/domains/suggestions/utils';
+import { stringifyDomainQueryObject } from './utils';
 
 export const getState = ( state: State ) => state;
-export const getDomainSuggestions = ( state: State, queryObject: DomainSuggestionQuery ) => {
-	const serializedQuery = getSerializedDomainsSuggestionsQuery( queryObject );
-	if ( serializedQuery ) {
-		return state.domainSuggestions[ serializedQuery ];
-	}
-	return null;
+
+type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery, 'query' > >;
+export const getDomainSuggestions = (
+	state: State,
+	search: string,
```

Comment:
Type as `DomainSuggestionQuery[ query ]`? (Not sure this exact syntax works. Also, might be over the top. Just thought single source of truth and all.)

Reply:
Yes, this could be:

```ts
	search: DomainSuggestionQuery[ 'query' ],
```

In terms of the external API (this selector is our external API), I tend to think that you do pass in "just a string", and internally we take care of normalizing it to the shape we expect. In that sense `string` is more appropriate.

The type system will help us to keep things consistent internally, so I'm not worried about the types getting out of sync.

I don't feel too strongly, however.